README.md

```diff
@@ -1,60 +1,82 @@
-
-  for (unsigned int i = 0; i < results.size(); i++) {
-    std::cout << keys[i] << ": " << results[i] << std::endl;
-  }
-}
+target_link_libraries(my_library cuco)
 ```
-Using a GPU-based map can be very beneficial when the workload includes some parallelism. As a result, `cuCollections::unordered_map` will accept, vectors of keys and values and insert them concurrently. This code snippet synthesizes dummy vectors for keys and values, then it inserts them in a `cuCollections::unordered_map`. Subsequently, it queries the map and prints the results which are returned in a vector called `results`. You may refer to the `examples` directory for more usage instances.
-
-## Integration
-`cuCollections` is designed to be added to projects as a submodule. In such settings, examples and benchmarks will not be build. As instructed in what follows, it is straightforward to integrate cuCollections in an existing project.
-- Add `cuCollections` as submodule: 
-```bash
-mkdir thirdparty && cd thirdparty
-git submodule add https://github.com/rapidsai/cuCollections.git cuCollections
-git submodule update --init --recursive
+
+This will take care of downloading `cuCollections` from GitHub and making the headers available in a location that can be found by CMake. Linking against the `cuco` target will provide everything needed for `cuco` to be used by the `my_library` target.
+
+[1] `cuCollections` is header-only and therefore there is no binary component to "link" against. The linking terminology comes from CMake's `target_link_libraries` which is still used even for header-only library targets. 
+
+## Dependencies
+- C++14
+- [libcu++](https://github.com/NVIDIA/libcudacxx)
+- [CUB](https://github.com/thrust/cub)
+- Volta+ 
+    - Pascal is partially supported. Any data structures that require blocking algorithms are not supported. See [libcu++](https://nvidia.github.io/libcudacxx/setup/requirements.html#device-architectures) documentation for more details.
+
+## Building cuCollections
+
+Since `cuCollections` is header-only, there is nothing to build to use it. 
+
+To build the tests, benchmarks, and examples:
+
+```
+cd $CUCO_ROOT
+mkdir -p build
+cd build
+cmake .. 
+make
 ```
-- Add `cuCollections` as a subdirectory to the `CMakeLists.txt`: `add_subdirectory("${CMAKE_SOURCE_DIR}/thirdparty/cuCollections")`.
-- Add `"${CU_COLLECTIONS_INCLUDE_DIRS}"` to your `include_directories`.
-- Include the related header file when applicable. For instance, to use concurrent maps, you may include `<cuHashmap.cuh>`.
+Binaries will be built into:
+- `build/tests/`
+- `build/gbenchmarks/`
+- `build/examples/`
+
+
+## Data Structures
+
+We plan to add many GPU-accelerated, concurrent data structures to `cuCollections`. As of now, the two flagships are variants of hash tables. 
+
+### `static_map`
+
+`cuco::static_map` is a fixed-size hash table using open addressing with linear probing. See the Doxygen documentation in `static_map.cuh` for more detailed information.
```

Comment:
I must say I'm not a big fan of moving all the documentation into doxygen. I know it's easier to keep stuff in one place, but for readability we could have a section here in README with an example of how static map is created and used, so users won't have to go into the source repo, find the file and the example code embedded into doxygen. We can still have a note that the latest and most recent usage is in doxygen, but at least build the high level picture for users right here in README. Also we should add some perf data - can we use Nico's results? One perf plot would be enough.

Reply:
Another idea is to post gbenchmarks "expected" results, say on V100?

Reply:
> I must say I'm not a big fan of moving all the documentation into doxygen 

This is a temporary state of affairs. In the future the Doxygen will be rendered ([like libcudf](https://docs.rapids.ai/api/libcudf/nightly/group__copy__gather.html)) so you don't have to go look into the source code.

Reply:
> Also we should add some perf data - can we use Nico's results? One perf plot would be enough.

I'm uneasy about adding a performance plot at this point. With so many optimizations and algorithmic ideas in flight, it would be a lot of effort to make sure that graph stays up to date. Furthermore, I wouldn't want people grabbing that graph and using it in formal performance comparisons.

Reply:
Doxygen is not going to replace a quick start guide on using the hash map in the code. If I’m a developer that just landed on the main readme page and interested in trying this out, I would be looking for a few lines of code showing how it works with some explanations of what’s going on. Something to highlight the key usage scenario. I’m curious why did you decide to remove the simple example from the updated version? Maybe instead just show what’s possible in the current hash map design? I envision that later when we have a blog or something like that, the main usage scenarios would be covered there, and so we could just point to that.

Reply:
cuco will have more than just hash maps, and it's not feasible to provide a quickstart for every data structure in the library, so I wanted to design the README to account for this. I added links to the `examples` directory at the top of the file, as well as data structure specific examples.